python/ray/serve/_private/common.py

```diff
@@ -118,17 +118,76 @@ class DeploymentStatusTrigger(str, Enum):
     DELETING = "DELETING"
 
 
+# Internal Enum used to manage deployment state machine
+class DeploymentStatusInternalTrigger(str, Enum):
+    HEALTHY = "HEALTHY"
+    CONFIG_UPDATE = "CONFIG_UPDATE"
+    AUTOSCALE_UP = "AUTOSCALE_UP"
+    AUTOSCALE_DOWN = "AUTOSCALE_DOWN"
+    MANUALLY_INCREASE_NUM_REPLICAS = "MANUALLY_INCREASE_NUM_REPLICAS"
+    MANUALLY_DECREASE_NUM_REPLICAS = "MANUALLY_DECREASE_NUM_REPLICAS"
+    REPLICA_STARTUP_FAILED = "REPLICA_STARTUP_FAILED"
+    HEALTH_CHECK_FAILED = "HEALTH_CHECK_FAILED"
+    INTERNAL_ERROR = "INTERNAL_ERROR"
+    DELETE = "DELETE"
+
+
 @dataclass(eq=True)
 class DeploymentStatusInfo:
     name: str
     status: DeploymentStatus
     status_trigger: DeploymentStatusTrigger
     message: str = ""
 
+    @staticmethod
+    def ranking_order() -> List[
+        Tuple[DeploymentStatus, Optional[DeploymentStatusTrigger]]
+    ]:
```

Comment:
Given this is fully static, why does it need to be a method? This will dynamically re-generate the list each time it's called. Would prefer to make it a class variable or global constant.

Reply:
Made it a global constant!